docs/content/en/docs/concepts/kube_context/_index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Kube-context activation"
+linkTitle: "Kube-context activation"
+weight: 80
+---
+
+This page discusses how Skaffold selects the kube-context.
+
+
+When interacting with a kubernetes cluster, Skaffold does so via a kube-context.
+Thus, the selected kube-context determines the kubernetes cluster, the kubernetes user, and the default namespace.
+By default, Skaffold uses the _current_ kube-context from your kube-config file.
+
+To override this default, Skaffold offers two options:
+
+1. Via the `kube-context` flag
+
+    ```bash
+    skaffold dev --kube-context <myrepo>
+    ```
+
+1. Via the `deploy.kubeContext` option in `skaffold.yaml`
+
+    ```yaml
+    deploy:
+      kubeContext: minikube
+    ```
+
+When both are given, the CLI flag always takes precedence.
+
+### Kube-context activation and Skaffold profiles
+
+The kube-context has a double role for Skaffold profiles:
+
+1. Profiles may be auto-activated by a given kube-context.
```

Comment:
I am not sure if this is true. 
If `kube-context` is minikube, then it does not mean we are activating the whole profile `profile-2` it can have other config overrides too.

Reply:
I am pretty sure that this is accurate. I even verified with an example.

Reply:
yep this is how it works.